src/components/StrongPasswordInput/index.js

```diff
@@ -0,0 +1,203 @@
+    /** The maximum number of characters allowed in the field. */
+    maxLength: PropTypes.number,
+    /** The minimum number of characters allowed in the field. */
+    minLength: PropTypes.number,
+    /** Shows the help message below the input. */
+    bottomHelpText: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
+    /** Specifies that an input field must be filled out before submitting the form.
+     * This value defaults to false. */
+    required: PropTypes.bool,
+    /** Specifies that an input field must be filled out before submitting the form. */
+    error: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
+    /** Specifies that an input element should be disabled. This value defaults to false. */
+    disabled: PropTypes.bool,
+    /** Specifies that an input field is read-only. This value defaults to false. */
+    readOnly: PropTypes.bool,
+    /** Specifies the tab order of an element (when the tab button is used for navigating). */
+    tabIndex: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+    /** The action triggered when a value attribute changes. */
+    onChange: PropTypes.func,
+    /** The action triggered when the element is clicked. */
+    onClick: PropTypes.func,
+    /** The action triggered when the element receives focus. */
+    onFocus: PropTypes.func,
+    /** The action triggered when the element releases focus. */
+    onBlur: PropTypes.func,
+    /** The action triggered when a key is pressed on the element. */
+    onKeyDown: PropTypes.func,
+    /** A CSS class for the outer element, in addition to the component's base classes. */
+    className: PropTypes.string,
+    /** An object with custom style applied to the outer element. */
+    style: PropTypes.object,
+    /** The id of the outer element. */
+    id: PropTypes.string,
+    /** The state of the password strenght */
+    passwordState: PropTypes.oneOf(['weak', 'average', 'strong']),
+    /** The label to show at the right of the bar */
+    passwordStateLabel: PropTypes.shape({
+        weak: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
+        average: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
+        strong: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
+    }),
+};
+
+StrongPasswordInput.defaultProps = {
+    value: undefined,
+    name: undefined,
+    placeholder: null,
+    icon: undefined,
+    iconPosition: 'left',
+    maxLength: undefined,
+    minLength: undefined,
+    bottomHelpText: null,
+    required: false,
+    error: null,
+    disabled: false,
+    readOnly: false,
+    onChange: () => {},
+    tabIndex: undefined,
+    onClick: () => {},
+    onFocus: () => {},
+    onBlur: () => {},
+    onKeyDown: () => {},
+    className: undefined,
+    style: undefined,
+    id: undefined,
+    label: undefined,
+    hideLabel: false,
+    passwordState: undefined,
+    passwordStateLabel: undefined,
+};
+
+export default StrongPasswordInput;
```

Comment:
since this is an inputable component and it can be used with redux-form we need to export it as other inputs:
```export default withReduxForm()```